linkcheck/checker/httpurl.py

```diff
@@ -78,15 +82,9 @@ def content_allows_robots (self):
         """
         if not self.is_html():
             return True
-        # construct handler object
-        handler = linkparse.MetaRobotsFinder()
-        # parse
-        try:
-            htmlsoup.process_soup(handler, self.get_soup())
-        except linkparse.StopParse as msg:
-            log.debug(LOG_CHECK, "Stopped parsing: %s", msg)
-            pass
-        return handler.follow
+
+        soup = self.get_soup()
+        return not soup.find("meta", attrs={"name": "robots"}, content=nofollow_re)
```

Comment:
I'm a bit thrown by the asymmetry between matching the `name` attribute and matching the `content` attribute.

Would either of these two alternatives work?

```suggestion
        return not soup.find("meta", attrs={"name": "robots", "content": nofollow_re})
```

```suggestion
        return not soup.find("meta", name="robots", content=nofollow_re)
```

?

Reply:
name is reserved by Beautiful Soup for its own use. Bit of making a point there...

Reply:
what about the other one?

Reply:
Yes, that works too. I've changed it to use it instead.